checkmanga.py

```diff
@@ -2,78 +2,128 @@
-    con.commit()
-
-def check_manga(cur, con, cache_dict, auto_up=True):
-    """Input: The cursor object and an auto_update boolean (default=True) that tells the function whether to automatically update LastChapterRead to be the same as MostRecentChapter.
-    Output: None. Updates the DB by scraping for chapters for manga where Status = 'Ongoing' and where MostRecentChapter = LastChapterRead"""
-    cur.execute("SELECT * FROM manga WHERE Status = 'Ongoing'")
-    all_rows = cur.fetchall()    
-    for entry in all_rows:
-        if cache_dict.get(entry["Site"]) != None:
-            site_path = cache_dict.get(entry["Site"])
-        else:
-        #The SQL statement below retrieves the "site_path" (aka instructions on how to scrape that manga site) from the tags table of the db.
-            cur.execute("SELECT * FROM tags WHERE site=(?) ORDER BY ID ASC", [entry['Site']])
-            site_path = cur.fetchall()
-            cache_dict[entry["Site"]] = site_path
-        print "Currently Scraping:", entry["Title"]
-        try:
-            most_recent_chapter = scrape_latest_chapter(entry['Url'], site_path)
-            if entry['MostRecentChapter'] != most_recent_chapter:     
-                cur.execute("UPDATE manga SET MostRecentChapter = (?) WHERE Title = (?)", (most_recent_chapter,entry["Title"])) 
-                print 'A new chapter has been released for ' + entry["Title"]
-            con.commit()
-        except requests.exceptions.ConnectionError:
-            print "Connection Timed out while scraping:",entry["Title"] ,"site may be down."
-        # if auto_up == True:
-        #     update_last_read(entry['Title'], cur, con)
-
-
-if __name__ == "__main__":
-    if len(sys.argv) == 1:
-        sys.exit("Please pass '--check' or '--bookmarks' as an argument to check manga.")
-    con = sqlite3.connect('mycheckmanga.db')
-    con.row_factory = sqlite3.Row
-    cur = con.cursor()
-    if (sys.argv[1]=='--check'):
-        cache_dict = {}
-        check_manga(cur, con, cache_dict)
-        con.close()
-    elif (sys.argv[1]== '--bookmarks'):
-        cur.execute("SELECT Title, LastChapterRead FROM manga")
-        for manga in cur.fetchall():
-            print manga 
-
-
-
-
-
-
+            #The SQL statement below retrieves the "site_path" (aka instructions on how to scrape that manga site) from the tags table of the db.
+            self.cur.execute("SELECT * FROM tags WHERE site=(?) ORDER BY ID ASC", [site])
+            site_path = self.cur.fetchall()
+            self.site_path_cache[site] = site_path
+        return site_path 
+    def add_manga(self, req):
+        """Input: A dictionary object representing request Json.
+        Output: Returns True after adding a new manga title to the manga table."""
+        if req["Site"] not in self.get_all_supported_sites():
+            return {"Error": "Requested site currently not supported by checkmanga."}
```

Comment:
not sure how all this works, but you should probably `raise BadRequestException` or something

Reply:
this should return 400